### PR TITLE
fix: sort Verify tool Species filter alphabetically

### LINKED ISSUE #208: List species in alphabetical order
The species listed in the Verify tool Species filter and tag selection appear in a seemingly random order.
They should be in alphabetical order.

image

image

The best place to addressed this is at the API request:
https://github.com/Greenstand/treetracker-admin-client/blob/ba9158f7e4be30644b6568dbd7b2c65ebc95a46c/src/api/treeTrackerApi.js#L164

### PR DESCRIPTION
**Pull request by bstetzer32**
## Description
Updated getSpecies query to sort the list of Species by name alphabetically for display on Verify tool.

**Issue(s) addressed**
- Resolves #208 

**What kind of change(s) does this PR introduce?** 
- [x] Enhancement
- [ ] Bug fix
- [ ] Refactor

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features) - not applicable
- [ ] Docs have been added / updated (for bug fixes / features) - not applicable

## Issue

**What is the current behavior?**
The list of Species for the Verify tool filter appears to load in random order. 

**What is the new behavior?**
The list of Species for the Verify tool filter loads in alphabetical order.

## Breaking change
**Does this PR introduce a breaking change?** 
No breaking changes.

### CODE REVIEW

File: src/api/treeTrackerApi.js [hunk 1]
```diff
@@ -163,16 +163,27 @@ export default {
...
+        headers: {
+          'content-type': 'application/json',
+          Authorization: session.token,
+        },
+        signal: abortController?.signal,
+      })
+        .then(handleResponse)
+        // Sort species list alphabetically by species name
+        .then((res) => {
+          res = res.sort((a, b) => {
```
Comment (nmcharlton) [inline src/api/treeTrackerApi.js:178]: Our API actually supports ordering in the query string, so there's no need to sort the results in the client:
```js
165:    const query = `${process.env.REACT_APP_API_ROOT}/api/species?filter[order]=name`;
```

**Review by nmcharlton — COMMENTED** [review]
Just one suggestion for a simpler implementation.

File: src/api/treeTrackerApi.js (hunk 1 above)
Comment (bstetzer32) [reply src/api/treeTrackerApi.js]: Thank you for your insight! I removed the client-side filtering, made the change to the query that you suggested, and merged the upstream changes from master into this branch

**Review by nmcharlton — APPROVED** [review]
Thanks @bstetzer32!

**Pull request merged**